prov/tcp/src/tcpx_ep.c

```diff
@@ -63,6 +63,27 @@ void tcpx_hdr_bswap(struct tcpx_base_hdr *hdr)
 		cur[i] = ntohll(cur[i]);
 }
 
+#ifdef MSG_ZEROCOPY
+static void tcpx_set_zerocopy(SOCKET sock)
+{
+	int val = 1;
+
+	(void) setsockopt(sock, SOL_SOCKET, SO_ZEROCOPY, &val, sizeof(val));
+}
+
+static void tcpx_config_bsock(struct ofi_bsock *bsock)
+{
+	int ret, val = 0;
+	socklen_t len = sizeof(val);
+
+	ret = getsockopt(bsock->sock, SOL_SOCKET, SO_ZEROCOPY, &val, &len);
+	if (!ret && val)
+		bsock->zerocopy_size = OFI_ZEROCOPY_SIZE;
```

Comment:
It seems that we cannot easily figure out if zerocopy is enabled.
It would be great if libfabric could return to the application what specific provider features are enabled for the endpoint (zero-copy, io_uring, etc...)

Reply:
My intent is to follow this with an option to configure it.  I can pass the provider to this, so we can print the result.

Reply:
Along the lines of your comment, I've had similar request before and have given it some thought.  The most reasonable option is to have the libfabric core handle this task somehow.  For example, capture the fi_info data used to open the fabric, domain, and endpoint.  And define some mechanism for the core to obtain the used value for any fi_param variables.  Anything beyond that becomes difficult and requires explicit opt-in by the providers.